### PR TITLE
New SparsePatternCref type

### PR DESCRIPTION
**Pull request by AnHeuermann**
## Changes
 - New types
    - `type SparsePatternCref = tuple< .DAE.ComponentRef, list< .DAE.ComponentRef>>`
    - `type SparsePatternCrefs = list<SparsePatternCref>`
    - To be used in BackEnd.
  - Changed printSparsityPattern to printSparsityPatternCrefs
  - Use new type in printSparsityPatternCrefs
  - Fixed typo and updated testsuite.

### CODE REVIEW

File: OMCompiler/Compiler/BackEnd/SymbolicJacobian.mo [hunk 1]
```diff
@@ -2021,7 +2021,7 @@ algorithm
         indepVars = createInDepVars(inDiffVars, false);
 
         if Flags.isSet(Flags.JAC_DUMP) then
-          print("Crete symbolic Jacobianis from:\n");
+          print("Create symbolic Jacobianis from:\n");
```
Comment (JKRT) [inline OMCompiler/Compiler/BackEnd/SymbolicJacobian.mo]: This should read Jacobians and not Jacobianis I think

File: OMCompiler/Compiler/BackEnd/BackendDAE.mo [hunk 2]
```diff
@@ -783,15 +783,18 @@ type SymbolicJacobian = tuple<BackendDAE,               // symbolic equation sys
                               list<Var>,                // diff vars (independent vars)
                               list<Var>,                // diffed vars (residual vars)
                               list<Var>,                // all diffed vars (residual vars + dependent vars)
-                              list< .DAE.ComponentRef>    // original dependent variables
+                              list< .DAE.ComponentRef>  // original dependent variables
                               >;
 
+type SparsePatternCref = tuple< .DAE.ComponentRef, list< .DAE.ComponentRef>>;
```
Comment (adrpo) [inline OMCompiler/Compiler/BackEnd/BackendDAE.mo]: You could probably name this somehow: list< .DAE.ComponentRef> and then use it in SparsePaternCref and also in SparsePattern.

File: testsuite/openmodelica/debugDumps/symjacdump.mos [hunk 3]
```diff
@@ -27,7 +27,7 @@ buildModel(testSymjacdump); getErrorString();
 // ""
 // true
 // ""
-// Crete symbolic Jacobianis from:
+// Create symbolic Jacobianis from:
```
Comment (JKRT) [inline testsuite/openmodelica/debugDumps/symjacdump.mos]: Same misspelling here as well

File: OMCompiler/Compiler/BackEnd/BackendDump.mo [hunk 4]
```diff
@@ -830,9 +817,9 @@ algorithm
   print("dependents [or outputs] (" + intString(listLength(diffedVars)) + ")\n");
```
Comment (JKRT) [inline OMCompiler/Compiler/BackEnd/BackendDump.mo]: Maybe start dependents with capital case

**Comment by adrpo** [comment]
The changes are fine to me, just some type naming and typos fixed.

File: OMCompiler/Compiler/BackEnd/BackendDAE.mo [hunk 5]
```diff
@@ -783,15 +783,18 @@ type SymbolicJacobian = tuple<BackendDAE,               // symbolic equation sys
...
+
 public
-type SparsePattern = tuple<list<tuple< .DAE.ComponentRef, list< .DAE.ComponentRef>>>, // column-wise sparse pattern
-                           list<tuple< .DAE.ComponentRef, list< .DAE.ComponentRef>>>, // row-wise sparse pattern
-                           tuple<list< .DAE.ComponentRef>,                            // diff vars
-                                 list< .DAE.ComponentRef>>,                           // diffed vars
-                           Integer>;                                                  // nonZeroElements
+type SparsePattern = tuple<SparsePatternCrefs,              // column-wise sparse pattern
+                           SparsePatternCrefs,              // row-wise sparse pattern
+                           tuple<list< .DAE.ComponentRef>,  // diff vars
```
Comment (JKRT) [inline OMCompiler/Compiler/BackEnd/BackendDAE.mo]: Maybe add a more concrete comment then diff vars

File: OMCompiler/Compiler/BackEnd/BackendDAE.mo (hunk 2 above)
Comment (AnHeuermann) [reply OMCompiler/Compiler/BackEnd/BackendDAE.mo]: You mean create a type for `list<DAE.ComponentRef>` in DAE.mo somewhere?

Comment (adrpo) [reply OMCompiler/Compiler/BackEnd/BackendDAE.mo]: That works also, you could call it CrefList or Crefs.

Comment (AnHeuermann) [reply OMCompiler/Compiler/BackEnd/BackendDAE.mo]: Changing `list<DAE.ComponentRef>` everywhere is not easy done automatically. It is used in a lot of places and for the bootstrapped compiler. I would prever to keep it this way for now.

Comment (adrpo) [reply OMCompiler/Compiler/BackEnd/BackendDAE.mo]: Is ok. We can do that later.

File: OMCompiler/Compiler/BackEnd/BackendDAE.mo (hunk 5 above)
Comment (AnHeuermann) [reply OMCompiler/Compiler/BackEnd/BackendDAE.mo]: I added some comment to make it more clear.

File: OMCompiler/Compiler/BackEnd/SymbolicJacobian.mo (hunk 1 above)
Comment (AnHeuermann) [reply OMCompiler/Compiler/BackEnd/SymbolicJacobian.mo]: At least I found 50%

File: OMCompiler/Compiler/BackEnd/BackendDump.mo (hunk 4 above)
Comment (AnHeuermann) [reply OMCompiler/Compiler/BackEnd/BackendDump.mo]: Updated it

**Comment by AnHeuermann** [comment]
@adrpo Would you merge the PR?

**Pull request merged**